fml/endianness.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_ENDIANNESS_H_
+#define FLUTTER_FML_ENDIANNESS_H_
+
+#include <cstdint>
+#include <type_traits>
+#if defined(_MSC_VER)
+#include "intrin.h"
+#endif
+
+#include "flutter/fml/build_config.h"
+
+// Compiler intrinsics for flipping endianness.
+#define BYTESWAP16(n) __builtin_bswap16(n)
```

Comment:
Please "namespace" macros in FML with the `FML_` prefix so we don't run into macro collisions. So, `FML_BYTESWAP_BLAH`.

Reply:
Done.